src/SSHDebugPS/Docker/DockerConnection.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier;
+
+namespace Microsoft.SSHDebugPS.Docker
+{
+    internal class DockerConnection : PipeConnection
+    {
+        private string _containerName;
+
+        public DockerConnection(DockerTransportSettings settings, Connection outerConnection, string name, string containerName)
+            : this(settings, outerConnection, name, containerName, null)
+        { }
+
+        public DockerConnection(DockerTransportSettings settings, Connection outerConnection, string name, string containerName, int? timeout)
+        : base(settings, outerConnection, name, timeout)
+        {
+            _containerName = containerName;
+        }
+
+        public override void BeginExecuteAsyncCommand(string commandText, bool runInShell, IDebugUnixShellCommandCallback callback, out IDebugUnixShellAsyncCommand asyncCommand)
+        {
+            if (IsClosed)
+            {
+                throw new ObjectDisposedException(nameof(PipeConnection));
+            }
+
+            if (runInShell)
+            {
+                AD7UnixAsyncShellCommand command = new AD7UnixAsyncShellCommand(CreateShellFromSettings(TransportSettings, OuterConnection), callback, closeShellOnComplete: true);
+                command.Start(commandText);
+
+                asyncCommand = command;
+            }
+            else
+            {
+                /// TODO: figure out what RunInShell means?
```

Comment:
>RunInShell [](start = 42, length = 10)

I added the `runInShell` parameter to allow consumers to run a command directly rather than having to start a shell and have the shell run the command to avoid the output mangling that sometimes entails.